bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandlerFactory.java

```diff
@@ -0,0 +1,374 @@
+        }
+        return false;
+    }
+
+    /**
+     * A programInfo or playStatus event was received from the receiver
+     *
+     * @param mrMac MR MAC address (used to map the device)
+     * @param jsonEvent Event data in JSON format
+     * @return true: thing handler was called, false: failed, e.g. unknown device
+     */
+    @SuppressWarnings("null")
+    public boolean notifyMREvent(String mrMac, String jsonEvent) {
+        try {
+            logger.trace("Received MR event from MAC {0}, JSON={1}", mrMac, jsonEvent);
+            MagentaTVDevice dev = lookupDevice(mrMac);
+            if ((dev != null) && (dev.thingHandler != null)) {
+                dev.thingHandler.onMREvent(jsonEvent);
+                return true;
+            }
+            logger.fatal("Received event for unregistered MR MAC address {0}, JSON={1}", mrMac);
+        } catch (RuntimeException e) {
+            logger.exception(e, "Unable to process MR event! json={0}", jsonEvent);
+        }
+        return false;
+    }
+
+    /**
+     * The PowerOff Listener got a byebye message. This comes in when the receiver
+     * was is going to suspend mode.
+     *
+     * @param ipAddress receiver IP
+     */
+    @SuppressWarnings("null")
+    public void onPowerOff(String ipAddress) {
+        try {
+            logger.debug("ByeBye message received for IP {0}", ipAddress);
+            MagentaTVDevice dev = lookupDevice(ipAddress);
+            if ((dev != null) && (dev.thingHandler != null)) {
+                logger.trace("Continue with UDN {0}", dev.udn);
+                dev.thingHandler.onPowerOff();
+            }
+        } catch (RuntimeException | MagentaTVException e) {
+            logger.exception(e, "Unable to process SSDP message for IP {0}", ipAddress);
+        }
+
+    }
+
+    /*
+     * @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
+     * public void setMagentaTVNotifyServlet(MagentaTVNotifyServlet servlet) {
+     * logger.debug("NotifyServlet bound");
+     * }
+     *
+     * public void unsetMagentaTVNotifyServlet(MagentaTVNotifyServlet servlet) {
+     * logger.debug("NotifyServlet unbound");
+     * }
+     */
+    @SuppressWarnings("null")
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
+    protected void setNetworkAddressService(NetworkAddressService networkAddressService) {
+        logger.debug("networkAddressService bound");
+        try {
+            logger.debug("Initialize network access");
+            network.initLocalNet(networkAddressService);
+            NetworkInterface intf = network.getLocalInterface();
+            upnpListener = new MagentaTVPoweroffListener(this, intf);
+            if (!upnpListener.isStarted()) {
+                upnpListener.start();
+            }
+        } catch (MagentaTVException e) {
+            logger.exception(e, "Unable to initialize network access");
```

Comment:
change to logger .debug

Reply:
important message, indicates that the binding is not initialized properly, can't detect local IP settings etc.; that's something the user should get aware of, otherwise communication does not work

Reply:
Then it should be level WARN